xlators/cluster/dht/src/dht-rebalance.c

```diff
@@ -2322,17 +2312,32 @@ rebalance_task(void *data)
     dht_local_t *local = NULL;
     call_frame_t *frame = NULL;
     int fop_errno = 0;
+    xlator_t *hashed_subvol;
+    xlator_t *cached_subvol;
 
     frame = data;
 
     local = frame->local;
 
     /* This function is 'synchrounous', hence if it returns,
        we are done with the task */
-    ret = dht_migrate_file(THIS, &local->loc, local->rebalance.from_subvol,
-                           local->rebalance.target_node, local->flags,
-                           &fop_errno);
 
+    hashed_subvol = local->rebalance.target_node;
```

Comment:
is hashed_subvol same as 'conf->hashed_subvol'?

I vaguely remember the issue of target being different from hashed_subvol when disks are getting full on that node, but there is space in other nodes. Please check that. If that is confirmed that `rebalance.target_node` is always same as hashed_subvol, then this makes sense.

Reply:
@amarts There is no functional change here, just a removal of unneeded operations.

Look at the code prioe to the changes - we pass local->rebalance.from_subvol and local->rebalance.target_node, so I just created local variables for these for readability.
In addition, in the code of `dht_migrate_file` we do:
```
if (from == to) {
        gf_msg_debug(this->name, 0,
                     "destination and source are same. file %s"
                     " might have migrated already",
                     loc->path);
        ret = 0;
        goto out;
    }
I moved this check out of `dht_migrate_file`.

In addition, at lint 1594 we used to do a subvol calculation, but this is not needed as we already have this information from the calling method `gf_defrag_migrate_single_file`.

Hope this clear things out. Let me know if you have doubts.

Reply:
:+1:  Ok, good from my end then.

Looks like you need to do a resubmit as there is conflict. Will approve after that.

Reply:
Yes, I'll address this shortly.